assets/js/blocks/products-by-attribute/block.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+import { BlockControls, InspectorControls } from '@wordpress/editor';
+import {
+	Button,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	Spinner,
+	Toolbar,
+	withSpokenMessages,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import { debounce } from 'lodash';
+import Gridicon from 'gridicons';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import getQuery from '../../utils/get-query';
+import ProductAttributeControl from '../../components/product-attribute-control';
+import ProductOrderbyControl from '../../components/product-orderby-control';
+import ProductPreview from '../../components/product-preview';
+
+/**
+ * Component to handle edit mode of "Products by Attribute".
+ */
+class ProductsByAttributeBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			products: [],
+			loaded: false,
+		};
+
+		this.debouncedGetProducts = debounce( this.getProducts.bind( this ), 200 );
+	}
+
+	componentDidMount() {
+		if ( this.props.attributes.attributes ) {
+			this.getProducts();
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		const hasChange = [
+			'attributes',
+			'attrOperator',
+			'columns',
+			'orderby',
+			'rows',
+		].reduce( ( acc, key ) => {
+			return acc || prevProps.attributes[ key ] !== this.props.attributes[ key ];
+		}, false );
+		if ( hasChange ) {
+			this.debouncedGetProducts();
+		}
+	}
+
+	getProducts() {
+		const blockAttributes = this.props.attributes;
+		if ( ! blockAttributes.attributes.length ) {
+			// We've removed all selected attributes, or no attributes have been selected yet.
+			this.setState( { products: [], loaded: true } );
+			return;
+		}
+		apiFetch( {
+			path: addQueryArgs(
+				'/wc-pb/v3/products',
+				getQuery( blockAttributes, this.props.name )
+			),
+		} )
+			.then( ( products ) => {
+				this.setState( { products, loaded: true } );
+			} )
+			.catch( () => {
+				this.setState( { products: [], loaded: true } );
```

Comment:
Probably, at the least I suppose we could log to the console if they're in dev mode. We use this pattern across all blocks though, so if we want to introduce any error logging it would be in another issue/PR.

Reply:
Added https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/380 to track this.